src/main/webapp/app/modules/canvas/visualization/visualization-properties/partials/properties/properties.tsx

```diff
@@ -0,0 +1,101 @@
+import { parseBool, parseString } from 'app/shared/util/common-utils.ts';
+
+export interface IPropertiesProps extends StateProps, DispatchProps {
+  features: readonly IFeature[];
+  visual: IVisualMetadataSet;
+  propstype: string;
+  property: Property;
+}
+
+const Properties = (props: IPropertiesProps) => {
+  const [property, setProperty] = useState([]);
+
+  const handleCheckboxChange = e => {
+    props.property.value = !props.property.value;
+    setProperty([props.property.value]);
+  };
+
+  const handleValueChange = (value, possibleValues = null) => {
+    if (possibleValues) {
+      const selectedValue = possibleValues.filter(item => {
+        return item.value === value;
+      });
+      props.property.value = selectedValue[0];
+    } else {
+      props.property.value = value;
+    }
+    setProperty([props.property.value]);
+  };
+
+  return (
+    <>
+      {props.property.type === 'NUMBER' && (
+        <TextField
+          type="NUMBER"
+          onChange={text => {
+            handleValueChange(text);
+          }}
+          value={props.property.value.toString() || ''}
+          label={props.property.propertyType.name}
+        />
+      )}
+      {props.property.type === 'SELECT' && (
+        <Picker
+          selectedKey={props.property.value['value']}
+          label={props.property.propertyType.name}
+          items={props.property.propertyType.possibleValues}
+          onSelectionChange={selected => handleValueChange(selected.toString(), props.property.propertyType.possibleValues)}
+        >
+          {item => <Item key={item.value}>{item.value}</Item>}
+        </Picker>
+      )}
+      {props.property.type === 'CHECKBOX' && (
+        <Switch
+          isEmphasized
+          onChange={() => {
+            handleCheckboxChange(props.property);
+          }}
+          isSelected={parseBool(props.property.value)}
+          defaultSelected={parseBool(props.property.propertyType.defaultValue)}
+        >
+          {props.property.propertyType.name}
+        </Switch>
+      )}
+      {props.property.type === 'COLOR_PICKER' && (
+        <TextField
+          onChange={text => {
+            handleValueChange(text);
+          }}
+          value={parseString(props.property.value)}
+          type="color"
+          label={props.property.propertyType.name}
+        />
+      )}
+      {props.property.type === 'TEXT' && (
+        <TextField
+          onChange={text => {
+            handleValueChange(text);
+          }}
+          value={parseString(props.property.value) || ''}
+          label={props.property.propertyType.name}
+        />
+      )}
+    </>
+  );
+};
+
+const mapStateToProps = (storeState: IRootState) => ({});
+
+const mapDispatchToProps = {};
+
+type StateProps = ReturnType<typeof mapStateToProps>;
+type DispatchProps = typeof mapDispatchToProps;
+
+export default connect(mapStateToProps, mapDispatchToProps)(Properties);
```

Comment:
map state and dispatch props are empty use a functional component

Reply:
okay